packages/react-api/src/Api.tsx

```diff
@@ -90,6 +91,24 @@ async function loadOnReady (api: ApiPromise): Promise<State> {
   });
   TokenUnit.setAbbr(tokenSymbol);
 
+  // Go through local storage and update the storage with old keyring value
+  const store = new BrowserStore();
+  store.all((key, json: any) => {
+    if(accountRegex.test(key) && json.encoding){
+      // The difference between new way of storing the keyring is only in the field content
+      // "encoding":{"content":["pkcs8",{"type":"sr25519"}] --- old
+      // "encoding":{"content":["pkcs8","sr25519"] --- new
+      // update the local storage with new way
+      const pkcs8 = json.encoding.content[0];
+      let accountType = json.encoding.content[1];
+      if (typeof accountType === 'object' && accountType !== null) {
```

Comment:
approach make sense to me!
can you add some unit tests to check bad input data cases and ed25519/sr25519 restore ok

Reply:
Ya.. I am trying to figure out, how it will be.. have started it.. will complete it tomorrow..